catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/ProjectUploadActivityTest.java

```diff
@@ -0,0 +1,204 @@
+import org.hamcrest.Matcher;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intending;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasExtras;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.common.Constants.IMAGE_DIRECTORY_NAME;
+import static org.catrobat.catroid.ui.recyclerview.dialog.UploadProgressDialogFragment.NUMBER_OF_UPLOADED_PROJECTS;
+import static org.catrobat.catroid.uiespresso.util.matchers.BundleMatchers.bundleHasMatchingString;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.StringContains.containsString;
+
+@RunWith(AndroidJUnit4.class)
+public class ProjectUploadActivityTest implements DeleteTestUserTask.OnDeleteTestUserCompleteListener {
+	@Rule
+	public BaseActivityInstrumentationRule<ProjectUploadActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ProjectUploadActivity.class);
+	String token = Constants.NO_TOKEN;
+	String projectName = "ProjectUploadActivityTest";
+	private String testUser = "testUser" + System.currentTimeMillis();
+	String testEmail = testUser + "@gmail.com";
+	private String testPassword = "pwspws";
+	private Matcher expectedIntent;
+
+	@Before
+	public void setUp() throws Exception {
+		ServerCalls.useTestUrl = true;
+		createProject(projectName);
+		FacebookSdk.sdkInitialize(InstrumentationRegistry.getTargetContext());
+
+		boolean userRegistered = ServerCalls.getInstance().register(testUser, testPassword, testEmail,
+				"de", "at", token, InstrumentationRegistry.getTargetContext());
+
+		assertTrue(userRegistered);
+
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		token = sharedPreferences.getString(Constants.TOKEN, Constants.NO_TOKEN);
+		boolean tokenOk = ServerCalls.getInstance().checkToken(token, testUser);
+		assertTrue(tokenOk);
+		Intent intent = new Intent(InstrumentationRegistry.getTargetContext(), ProjectUploadActivity.class)
+				.putExtra(ProjectUploadActivity.PROJECT_NAME, Utils.getCurrentProjectName(InstrumentationRegistry.getTargetContext()));
+
+		// to avoid showing RatePocketCodeDialogFragment during the test
+		sharedPreferences.edit()
```

Comment:
you should reset this change in `tearDown()`. Because it is good practice to leave the state of the application (and its shared preferences as you found them). So you could add a member to the test where you buffer the previous value of the preference, such as:

```
private int sharedPreferenceBuffer;

@Before
private void setUp() {
    sharedPreferenceBuffer = sharedPreferences.getInt(NUMBER_OF_UPLOADED_PROJECTS);
    ...
}

@After
private void tearDown() {
    sharedPreferences.putInt(NUMBER_OF_UPLOADED_PROJECTS, sharedPreferenceBuffer)
                     .commit();
   ...